sympy/combinatorics/schur_number.py

```diff
@@ -0,0 +1,120 @@
+"""
+The Schur number S(k) is the largest integer n for which the interval [1,n]
+can be partitioned into k sum-free sets.(http://mathworld.wolfram.com/SchurNumber.html)
+"""
+import math
+from sympy import Symbol
+from sympy.core import S
+from sympy.core.function import Function
+
+
+class schur_number_lower_bound(Function):
+    """
+    Gives the value for 1/2 * (3**n  - 1)
+    """
+
+    @classmethod
+    def eval(cls, k):
+        expr = (1/2) * (3**k - 1)
+
+        if k.is_Number:
+            if k is S.Infinity:
+                return S.Infinity
+            if k.is_zero:
+                return 0
+            if not k.is_Integer or k.is_negative:
+                raise ValueError("k should be a positive integer")
+            expr = int(expr)
+
+        return expr
+
+
+class schur_number_subsets_lower_bound(Function):
+    """
+    This function returns a lower bound to Schur's Number
+    """
+
+    @classmethod
+    def eval(cls, n):
+        n = int(n)
+
+        if n is S.Infinity:
+            return S.Infinity
+        if n <= 0:
+            raise ValueError("n must be a positive integer.")
+        elif n <= 3:
+            min_k = 1
+        else:
+            min_k = math.ceil(math.log(2*n + 1, 3))
+
+        return min_k
+
+def schur_partition(n):
+    """
+    This function makes the partition in the minimum number of sum-free subsets
```

Comment:
This docstring needs to follow the formatting outlined in the style guide https://docs.sympy.org/latest/documentation-style-guide.html

Reply:
This description does not make it clear at all what this function returns. Does it return subsets, or a number. If subsets, what form are they?

Reply:
It returns the subsets in a list of lists format and it is mentioned in the second sentence of the explanation. I will add new lines between the first and the second sentences to make it look clear.